tools/config/go-templates/model_simple.mustache

```diff
@@ -28,15 +28,8 @@ type {{classname}} struct {
 {{/deprecated}}
 	{{name}} {{^required}}{{^isFreeFormObject}}*{{/isFreeFormObject}}{{/required}}{{{dataType}}} `json:"{{baseName}}{{^required}},omitempty{{/required}}"{{#withXml}} xml:"{{baseName}}{{#isXmlAttribute}},attr{{/isXmlAttribute}}"{{/withXml}}{{#vendorExtensions.x-go-custom-tag}} {{{.}}}{{/vendorExtensions.x-go-custom-tag}}`
 {{/vars}}
-{{#isAdditionalPropertiesTrue}}
```

Comment:
these isAdditionalPropertiesTrue blocks seem not to have any effect

Reply:
Until they appear in the OpenAPI

Reply:
What you see is conicidence that OpenAPI input does not trigger some of those claims. If that changes entire logic of SDK will be broken and we would need to investigate why. 
If your approach is to make SDK lean let's propose it separately as there would be some extra work required including validation for those changes to not break SDK in the future.

Reply:
there are 10 occurrences of "additionalProperties" in transformed OpenAPI file, e.g. [here](https://github.com/mongodb/atlas-sdk-go/blob/main/openapi/atlas-api-transformed.yaml#L22082)

So not sure if this logic is working fine or it's for other scenarios that are not happening yet. 

I don't think logic would break but that new scenario would be ignored and wouldn't be taken into account.

Anyway I'm thinking about pausing this PR and see first if it's feasible to upgrade generator version.

Reply:
SGTM